src/include/parallel/task.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025, Sirius Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "duckdb/common/shared_ptr.hpp"
+#include "duckdb/common/unique_ptr.hpp"
+
+namespace sirius {
+namespace parallel {
+
+/**
+ * Interface for concrete task local states.
+ */
+class ITaskLocalState {
+public:
+  virtual ~ITaskLocalState() = default;
+};
+
+/**
+ * Interface for concrete task global states.
+ */
+class ITaskGlobalState {
+public:
+  virtual ~ITaskGlobalState() = default;
+};
+
+/**
+ * Interface for concrete executor tasks.
+ */
+class ITask {
+public:
+  ITask(duckdb::unique_ptr<ITaskLocalState> local_state, duckdb::shared_ptr<ITaskGlobalState> global_state)
```

Comment:
if we want this engine to be applicable to multiple runtimes sometime in the future wouldn't it be better to rely on the std implementations?

Reply:
I heard at some point we want to support something like Doris. The checks don't seem incredibly helpfpul to me particularly if you can ensure that the value is not null before hand.